### ISSUE TITLE
 Failed setup, will retry: '>' not supported between instances of 'NoneType' and 'datetime.datetime' 

### ISSUE THREAD

**Issue by 0utc45t** [mawinkler/astroweather#63]
Astroweather stopped working after upgrading of the integration a few days ago. I can't get it to work again even with downgrading to previous version. Or by removing and reinstalling integration. I used HACS to install Astroweather (and lovelace card). Debug log attached. My homeassistant is running on Yellow, everything is updated to newest versions available.

[home-assistant_astroweather_2024-10-09T11-50-57.696Z.log](https://github.com/user-attachments/files/17307615/home-assistant_astroweather_2024-10-09T11-50-57.696Z.log)

**Comment by mawinkler** [comment]
I found the problem. There are missing checks if the Moon either does not rise or set in a give timespan. Is it possible for you to send me your geographical location (long, lat, elevation) for validation?

**Comment by 0utc45t** [comment]
Hi,
These coordinates are close, but not exact, taken from Google maps.
Altitude around 12 meters.
65.064717,25.553043

BR,
Sami

**Comment by mawinkler** [comment]
Thanks to Finland.
Can you change the version of pyastroweatherio in the manifest.json of astroweather to `0.61.0.2` and give it a try?

```json
    "requirements": [
        "pyastroweatherio==0.61.0.2"
    ],
```

**Comment by 0utc45t** [comment]
I changed the version as requested and rebooted HA Yellow. Still not working and errors changed in the log file. There seems to be lot of warnings of other integrations as well... I'm not cutting away anything, so I don't accidentally remove anything you actually need to see. 

[home-assistant_2024-10-10T04-52-18.474Z.log](https://github.com/user-attachments/files/17322248/home-assistant_2024-10-10T04-52-18.474Z.log)

**Comment by mawinkler** [comment]
I released [0.61.1](https://github.com/mawinkler/astroweather/releases/tag/v0.61.1) which hopefully fixes your issues

**Comment by 0utc45t** [comment]
Yes, it seems it did. Thank you very much! :-D

**Comment by mawinkler** [comment]
fixed with [0.61.1](https://github.com/mawinkler/astroweather/releases/tag/v0.61.1)
